src/authentication/components/ExecutionConfigModal.tsx

```diff
@@ -0,0 +1,215 @@
+import React, { useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useForm } from "react-hook-form";
+import {
+  ActionGroup,
+  AlertVariant,
+  Button,
+  ButtonVariant,
+  Form,
+  FormGroup,
+  Modal,
+  ModalVariant,
+  TextInput,
+  ValidatedOptions,
+} from "@patternfly/react-core";
+import { CogIcon, TrashIcon } from "@patternfly/react-icons";
+
+import type AuthenticatorConfigRepresentation from "keycloak-admin/lib/defs/authenticatorConfigRepresentation";
+import type AuthenticatorConfigInfoRepresentation from "keycloak-admin/lib/defs/authenticatorConfigInfoRepresentation";
+import type { ConfigPropertyRepresentation } from "keycloak-admin/lib/defs/authenticatorConfigInfoRepresentation";
+import type { ExpandableExecution } from "../execution-model";
+import { useAdminClient, useFetch } from "../../context/auth/AdminClient";
+import { useAlerts } from "../../components/alert/Alerts";
+import { HelpItem } from "../../components/help-enabler/HelpItem";
+
+export const ExecutionConfigModal = ({
+  execution,
+}: {
+  execution: ExpandableExecution;
+}) => {
+  const { t } = useTranslation("authentication");
+  const adminClient = useAdminClient();
+  const { addAlert, addError } = useAlerts();
+
+  const [show, setShow] = useState(false);
+  const [config, setConfig] = useState<AuthenticatorConfigRepresentation>();
+  const [configDescription, setConfigDescription] =
+    useState<AuthenticatorConfigInfoRepresentation>();
+
+  const { register, errors, setValue, handleSubmit } = useForm();
+
+  const setupForm = (
+    configDescription: any,
+    config?: AuthenticatorConfigRepresentation
+  ) => {
+    configDescription.properties.map(
+      (property: ConfigPropertyRepresentation) => {
+        setValue(
+          property.name!,
+          config?.config?.[property.name!] || property.defaultValue || ""
+        );
+      }
+    );
+    if (config) {
+      setValue("alias", config.alias);
+      setValue("id", config.id);
+    }
+  };
+
+  useFetch(
+    async () => {
+      const configDescription =
+        await adminClient.authenticationManagement.getConfigDescription({
+          providerId: execution.providerId!,
+        });
+      let config: AuthenticatorConfigRepresentation | undefined;
+      if (execution.authenticationConfig) {
+        config = await adminClient.authenticationManagement.getConfig({
+          id: execution.authenticationConfig,
+        });
+      }
+      return { configDescription, config };
+    },
+    ({ configDescription, config }) => {
+      setConfigDescription(configDescription);
+      setConfig(config);
+    },
+    []
+  );
+
+  useEffect(() => {
+    if (configDescription) setupForm(configDescription, config), [open];
```

Comment:
There seems to be an array literal here that doesn't do anything, I think it can be removed.
```suggestion
    if (configDescription) setupForm(configDescription, config);
```

Reply:
it's the useEffect dependency I fixed it